extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -223,7 +223,7 @@ export abstract class AzureAuth implements vscode.Disposable {
 		}
 		// Let's try to convert the access token type, worst case we'll have to prompt the user to do an interactive authentication.
 		const result = await this.refreshToken(tenant, resource, baseTokens.refreshToken);
-		if (result.accessToken) {
+		if (result?.accessToken) {
```

Comment:
Please also : 

1. Update `refreshToken` to have the correct return type (`| undefined`)
2. Update/add docs to the relevant functions describing the behavior and what to expect. Be sure you look through the different scenarios that can result in undefined - for example exactly how/why user interaction may not be required
3. Fix line 201 to also account for the possibility of undefined being returned